File: src/test/java/koorkah/TestGeneral.java
package koorkah;

import static org.junit.jupiter.api.Assertions.assertThrows;
import static org.junit.jupiter.api.Assertions.assertTrue;

import org.junit.jupiter.api.Test;

 
class TestGeneral extends BaseT
{
    @Test
    void testUnknownToken()
    {
        Exception exception = assertThrows(RuntimeException.class, () ->       
            koorkah.toRegEx("green"));
        assertTrue(exception.getMessage().contains("token recognition error at: 'green'"));
    }
    
    @Test
    void testExample()
    {
          
        String expression = koorkah.toRegEx("bol capture{ somespace `cat` somespace } eol");
        assertRegexIs("^(\\p{Space}+cat\\p{Space}+)$", expression);
        assertRegexMatchesText(expression, " cat ");
        assertRegexMatchesText(expression, "   cat   ");
        assertRegexDoesntMatchText(expression, "cat  "); 
        assertRegexDoesntMatchText(expression, "   cat"); 
    }


}

File: src/python/test/TestGeneral.py
import unittest
from Utils import Utils 
from Refi import Refi
 
class TestGeneral(unittest.TestCase):

    def setUp(self):
        self.refi = Refi()
        self.utils = Utils()

    def test_UnknownToken(self):

        with self.assertRaises(Exception):
            self.refi.toRegEx("green")
    
    def test_Example(self):
          
        expression = self.refi.toRegEx("bol capture{ somespace `cat` somespace } eol")
        self.utils.assertRegexIs(r"^(\p{Space}+cat\p{Space}+)$", expression)
        self.utils.assertRegexMatchesText(expression, " cat ")
        self.utils.assertRegexMatchesText(expression, "   cat   ")
        self.utils.assertRegexDoesntMatchText(expression, "cat  ") 
        self.utils.assertRegexDoesntMatchText(expression, "   cat") 


    def runTest(self):
        pass  

if __name__ == '__main__':
    unittest.main()    
    


